src/litdata/utilities/encryption.py

```diff
@@ -0,0 +1,137 @@
+import base64
+import os
+from abc import ABC, abstractmethod
+
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import padding, rsa
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+
+class Encryption(ABC):
+    """Base class for encryption algorithm."""
+
+    @abstractmethod
+    def encrypt(self, data: bytes) -> bytes:
+        pass
+
+    @abstractmethod
+    def decrypt(self, data: bytes) -> bytes:
```

Comment:
Let's add `state_dict` method, so we can store the encryption state into the index.json for restoration.

Reply:
Sure

Reply:
I will look into  how this state_dict is working and then add the state_dict method